apps/web/src/app/recovery/reset-password/request/page.tsx

```diff
@@ -0,0 +1,74 @@
+'use client';
+
+import { Button } from '@/src/components/shadcn-ui/button';
+import { Input } from '@/src/components/shadcn-ui/input';
+import { useRouter } from 'next/navigation';
+import { platform } from '@/src/lib/trpc';
+import { useState } from 'react';
+import { toast } from 'sonner';
+
+export default function RequestResetPage() {
+  const [username, setUsername] = useState('');
+  const [email, setEmail] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
+  const router = useRouter();
+
+  const { mutateAsync: sendRecoveryEmail } =
+    platform.account.security.sendRecoveryEmail.useMutation();
+
+  const handleSubmit = async (e: React.FormEvent) => {
+    e.preventDefault();
+    setIsLoading(true);
+    try {
+      await sendRecoveryEmail({ username, email });
+      toast.success(
+        'Recovery email sent. Please check your inbox for the verification code.'
+      );
+      router.push('/recovery/reset-password/verify');
+    } catch (error) {
+      toast.error('An error occurred. Please try again.');
+    } finally {
+      setIsLoading(false);
+    }
+  };
```

Comment:
make handleSumbit a inline function on the on Submit event, put the the success, error, loading state on the `useMutation` call

```tsx
onSubmit={(e)=>form.handleSubmit(values=>{...})(e)}
```

Reply:
what's the reasoning behind this ?

Reply:
update: will not resolve on this PR let's discuss.

Reply:
I assumed that you would use form hooks, but as you are not going to do that. it's fine.

The handle submit will not have 3 lines of code only.
```ts
e.preventDefault();
e.stopPropagation();
await sendRecoveryEmail();
```

so you don't need a separate handleSubmit function, just use an arrow function on the form itself